modules/nlu/src/backend/engine.ts

```diff
@@ -113,11 +114,32 @@ export default class Engine implements NLUEngine {
       ctxToTrain
     }
 
-    // Model should be build here, Trainer should not have any idea of how this is stored
-    // Error handling should be done here
-    let model = await Trainer(input, Engine.tools)
+    let model: Partial<Model> = {
+      startedAt: new Date(),
+      languageCode: input.languageCode,
+      data: {
+        input
+      }
+    }
+
+    try {
+      const artefacts = await Trainer(input, Engine.tools)
+      model.success = true
+      _.merge(model, { success: true, data: { artefacts } })
+    } catch (err) {
+      model.success = false
+
+      if (err instanceof TrainingCanceledError) {
+        trainDebug.forBot(input.botId, 'Training aborted')
+      } else {
+        trainDebug.forBot(input.botId, `Could not finish training NLU model \n ${err}`)
```

Comment:
the TODO comment you've erased stated that you should use the bp.logger from the sdk in this catch

Reply:
Removed `input.botId` and added the TrainingCandeledError in the training-pipeline, the else will bubble up other errors